pkg/kubectl/cmd/login.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+)
+
+func NewCmdLogin(f *cmdutil.Factory, pathOptions *clientcmd.PathOptions, out io.Writer) *cobra.Command {
+
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "login Log in to the cluster.",
+		Long:  `login obtains the neccesary credentials to make authenticated requests to the cluster. The type of log in depends on the 'auth-provider' type in your configuration, and will likely require some type of user interaction.`,
```

Comment:
There are two parts of login extension.  

1. Baked in login code that could init the auth provider if a user could indicate it - i.e.  a flag like `--oidc`
2. A set of code that is not baked into login, but which could be added via a package installer such that kubectl login could delegate - i.e. a binary `kubectl-login-oidc` in the path that the flag `--oidc` would instruct `kubectl login --oidc` to exec with any shim logic in place.

I'd like to see both - bootstrapping a kube config initially is ok as a starting point, but I'd much rather be able to say `kubectl login --kerberos` or `kubectl login --gcloud`.